Sources/ConcurrencyHelpers/Event.swift

```diff
@@ -0,0 +1,93 @@
+import Atomics
+
+/**
+ * Synchronization primitive modelled after Event from WinAPI.
+ */
+public final class Event {
+    // Protects continuations array and write to signaled flag.
+    private var lock = Lock()
+
+    // Can be accessed for read without taking lock
+    private var signaled: UnsafeAtomic<Bool>
+
+    private typealias Continuation = CheckedContinuation<Void, Never>
+
+    private var continuations = [Continuation]()
+
+    /**
+     * Initialize a new event.
+     *
+     * - argument signaled: Initial state of event. Non-signaled by default.
+     */
+    public init(signaled: Bool = false) {
+        self.signaled = .create(signaled)
+    }
+
+    /**
+     * Wait for event to become signaled.
+     */
+    public func wait() async {
+        guard !isSignaled else {
+            return
+        }
+
+        lock.lock()
+
+        guard !isSignaled else {
+            lock.unlock()
```

Comment:
probably, better to use pattern as in library:
```
defer {
    lock.unlock()
}
```

Reply:
It's not possible, we need to release lock in the closure passed to `withCheckedContinuation`

Reply:
Could that closure be executed in other thread?
If so, is it allowed to release lock in other thread?

Reply:
I don't think so.